noir-projects/aztec-nr/aztec/src/context/private_context.nr

```diff
@@ -224,6 +226,30 @@ impl PrivateContext {
         keys.app_nullifier_secret_key
     }
 
+    // TODO: (#6176) Replace request_app_nullifier_secret_key above with this once we no longer get app nullifier secret key with address
+    pub fn request_app_nullifier_secret_key_with_master_public_nullifying_key_hash(&mut self, master_public_nullifying_key_hash: Field) -> Field {
+        let keys = if self.nullifier_key.is_none() {
+            let keys = get_nullifier_keys_with_master_nullifier_public_key_hash(master_public_nullifying_key_hash);
+            assert_eq(poseidon2_hash(keys.master_nullifier_public_key.serialize()), master_public_nullifying_key_hash);
+            let request = NullifierKeyValidationRequest {
+                master_nullifier_public_key: keys.master_nullifier_public_key,
+                app_nullifier_secret_key: keys.app_nullifier_secret_key
+            };
+            self.nullifier_key_validation_requests.push(request);
+            self.nullifier_key = Option::some(keys);
+            keys
+        } else {
+            let keys = self.nullifier_key.unwrap_unchecked();
+            // If MAX_NULLIFIER_KEY_VALIDATION_REQUESTS_PER_CALL is larger than 1, need to update the way the key pair is cached.
+            assert(MAX_NULLIFIER_KEY_VALIDATION_REQUESTS_PER_CALL == 1);
+            keys
+        };
+
+        // TODO: (#6176) (ek) Enable this after we nuke the account in all notes and when NullifierKeys has a reference to master_nullifier_public_key_hash
+        // assert_eq(keys.master_nullifier_public_key_hash, master_public_nullifying_key_hash);
```

Comment:
As mentioned in the comment above, this one seems useless actually, as it is really the same think we are checking in the two, one is saying 
- check key hash is x 
- other compute key hash and check is x.